### PR TITLE
Support correlated scalar subqueries (v2)

### PR DESCRIPTION
**Pull request by kokosing**
Support correlated count scalar subqueries

A scalar subquery is a simple query which aggregates all the rows into a
single group (empty GROUP BY clause). For example:

SELECT min(a) FROM t WHERE b > 0
This commit add support for correlated count scalar subqueries which
have WHERE cluase of conjunction of equality comparison. For example:

SELECT (SELECT min(a) FROM t1(a,c) WHERE b = 0 AND c = d) FROM t2(b,d)
Logically such query is rewritten into left outer join where correlation
used in eqality comparisons (from WHERE cluase) are used as join
condition.

### CODE REVIEW

**Comment by kokosing** [comment]
Supersedes https://github.com/prestodb/presto/pull/5737 and depends on #5687 (which depends on #5686), #5423 and https://github.com/prestodb/presto/pull/5823.

File: presto-docs/src/main/sphinx/sql/select.rst [hunk 1]
```diff
@@ -646,3 +646,54 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
...
+
+.. _exists:
+
+EXISTS
+^^^^^^
+
+.. code-block:: sql
+
+    SELECT nation.name, region.name
+    FROM nation n,
```
Comment (KBP-TDC) [inline presto-docs/src/main/sphinx/sql/select.rst]: Is comma after "n" needed? See more instances below

**Comment by kokosing** [comment]
Rebased.

**Comment by martint** [comment]
Dependencies are merged. Can you rebase?

**Comment by kokosing** [comment]
Superseded by #5984

**Pull request closed without merging**